frontend/src/pages/user/admin/allCoupons/[adminCouponID].tsx

```diff
@@ -0,0 +1,196 @@
+import { useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { Row, Col } from 'react-bootstrap';
+import DatePicker from 'react-datepicker';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faPen, faTrash } from '@fortawesome/free-solid-svg-icons';
+import 'react-datepicker/dist/react-datepicker.css';
+
+import TButton from '@components/TButton';
+import InfoItem from '@components/InfoItem';
+import CouponItem from '@components/CouponItem';
+import couponData from '@pages/coupon/couponData.json';
+
+interface Props {
+  id: number | null;
+  name: string;
+  policy: string;
+  date: string;
+  introduction: string;
+  tags: { name: string }[];
+}
+
+const EachAdminCoupon = () => {
+  const params = useParams();
+  const id = params.coupon_id;
+
+  const data: Props = {
+    id: null,
+    name: '',
+    policy: '',
+    date: '',
+    introduction: '',
+    tags: [],
+  };
+
+  const foundCoupon = couponData.find((coupon) => coupon.id.toString() === id);
```

Comment:
Could `coupon.id.toString` fail on `null`?

Reply:
The ``coupon.id.toString`` will come from api, so it shouldn't be null ?